src/tabs/Curve/animation_canvas_3d_curve.tsx

```diff
@@ -1,354 +1,334 @@
+   */
+  private onTimeSliderRelease = () => {
+    this.setState(
+      (prev) => ({
+        isPlaying: prev.wasPlaying,
+      }),
+      () => {
+        if (!this.state.isPlaying) {
+          this.callbackTimestamp = null;
+        } else {
+          this.reqFrame();
+        }
+      },
+    );
+  };
+
+  private onAngleSliderChange = (newAngle: number) => {
+    this.setState(
+      {
+        displayAngle: newAngle,
+      },
+      () => {
+        this.props.animation.angle = newAngle;
+        if (this.state.isPlaying) this.reqFrame();
+        else this.drawFrame();
+      },
+    );
+  };
+
+  /**
+   * Auto loop switch onChange callback
+   */
+  private onSwitchChange = () => {
+    this.setState((prev) => ({
+      isAutoLooping: !prev.isAutoLooping,
+    }));
+  };
+
+  public componentDidMount() {
+    this.drawFrame();
+  }
+
+  public render() {
+    return <div
+      style={{
+        width: '100%',
+      }}
+    >
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'center',
+        }}
+      >
+        <div
+          style={{
+            display: 'flex',
+            alignItems: 'center',
+            gap: BP_TAB_BUTTON_MARGIN,
+
+            width: '100%',
+            maxWidth: CANVAS_MAX_WIDTH,
+
+            paddingTop: BP_TEXT_MARGIN,
+            paddingBottom: BP_TEXT_MARGIN,
+          }}
+        >
+          <PlayButton
+            isPlaying={ this.state.isPlaying }
+            onClickCallback={ this.onPlayButtonClick }
+          />
+          <Tooltip2
+            content="Reset"
+            placement="top"
+          >
+            <Button onClick={ this.onResetButtonClick }>
+              <Icon icon={ IconNames.RESET } />
+            </Button>
+          </Tooltip2>
+          <div
+            style={{
+              display: 'flex',
+              flexDirection: 'column',
+              gap: BP_TEXT_MARGIN,
+
+              width: '100%',
+            }}
+          >
+            <Slider
+              value={ this.state.animTimestamp }
+              min={ 0 }
+              max={ this.animationDuration }
+              stepSize={ 1 }
+
+              labelRenderer={ false }
+
+              onChange={ this.onTimeSliderChange }
+              onRelease={ this.onTimeSliderRelease }
+            />
+            <Tooltip2
+              content="Display Angle"
+              placement="top"
+            >
+              <Slider
+                value={ this.state.displayAngle }
+                min={ 0 }
+                max={ 2 * Math.PI }
+                stepSize={ 0.01 }
+
+                labelRenderer={ false }
+
+                onChange={ this.onAngleSliderChange }
+              />
+            </Tooltip2>
+          </div>
+          <AutoLoopSwitch
+            isAutoLooping={ this.state.isAutoLooping }
+            onChangeCallback={ this.onSwitchChange }
+          />
+        </div>
+      </div>
+      <div
+        style={{
```

Comment:
I have included a margin at the top of this element previously (or it can be added to the controls as well) so that there is some space between the controls and the canvas render. Is that no longer an issue now?

Reply:
I did note the addition of marginTop to the controls bar in #223. However, when the canvases are in a MultiItemDisplay, that takes up even more vertical space and as is, has the potential to push the bottom of the canvas out of view on some typical window sizes (note the scroll bar):

![image](https://github.com/source-academy/modules/assets/44526554/eb065ad4-555a-4456-940e-141a2d15eac4)

Given this, I felt that the spacing without another margin was sufficient.

In a similar vein, I removed the side padding of the controls bar in an attempt to help it fit better within the narrow width of mobile devices, and keep some of the "wider sidebar" feel of the most recent merged changes. Though it currently doesn't fit that well on mobile, with the sliders seeming to insist on a minimum width:

![image](https://github.com/source-academy/modules/assets/44526554/1e5dc3ab-de49-496a-ae82-a337f8520298)

I did try playing with some of the styling such as text wrapping, but that came with negative side effects for certain widths of desktop view, so I ultimately scrapped those.

MultiItemDisplay actually adds its own 40px of horizontal padding all the time, even when there's only one item. Maybe removing all of its enforced spacings would help alleviate this issue? Then it would be up to each consuming component to decide on what spacing they wish to add, if any.

![image](https://github.com/source-academy/modules/assets/44526554/4354a8e8-bd30-4dad-baff-cb13909ca2a4)

Reply:
I have pushed a commit to remove the enforced spacing of MultiItemDisplay.